api/src/main/java/org/wildfly/transformer/ArchiveTransformer.java

```diff
@@ -13,76 +13,68 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.wildfly.transformer.tool.api;
+package org.wildfly.transformer;
 
 import java.io.Closeable;
 import java.io.File;
-import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.InputStream;
-import java.io.OutputStream;
 import java.io.IOException;
-import java.nio.file.Path;
+import java.io.OutputStream;
 import java.util.Calendar;
 import java.util.Enumeration;
 import java.util.Map;
 import java.util.jar.JarEntry;
 import java.util.jar.JarFile;
 import java.util.jar.JarOutputStream;
 
-import org.wildfly.transformer.Transformer;
-import org.wildfly.transformer.Transformer.Resource;
-import org.wildfly.transformer.TransformerBuilder;
-import org.wildfly.transformer.TransformerFactory;
-
 /**
- * Command line tool for transforming class files or jar files.
+ * Archive transformer tries to convert given archives to another archives by applying configured transformation rules.
+ * Can be used concurrently by multiple threads as instances of this class are thread safe.
```

Comment:
Does this depend on whether the underlying transformer is also thread safe?  I'm not sure if Eclipse Transformer is thread safe.

Reply:
I would guess it is thread safe @scottmarlow - I will review it.

Reply:
Eclipse transformer isn't thread safe AFAICS. However that is not a problem at all because this PR also changes
integration points. Since now on only single archive can be transformed one by one. This change doesn't require thread safety requirements. Before we were considering thread safety is important because we planned to support usecase where tools user could transform whole directory recursively and concurrently. This use case is not allowed since this PR will be applied. I will change javadoc accordingly.